### ISSUE TITLE
@Composable onDispose is not called if we close the window

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-jb#150]
```
import androidx.compose.desktop.Window
import androidx.compose.foundation.layout.ExperimentalLayout
import androidx.compose.runtime.onDispose

@OptIn(ExperimentalLayout::class)
fun main() = Window {
    onDispose {
       println("onDispose")
    }
}
```

I think we need to dispose the composition inside ComposeWindow/ComposePanel:
```
fun ComposeWindow.setContent(content: @Composable () -> Unit) {
    composition = this.layer.setContent(
        parent = parent,
        content = content
    )
    return composition
}

override fun dispose() {
    composition.dispose()
    layer.dispose()
    super.dispose()
}
```

**Comment by Rsedaikin** [comment]
Fixed in version `0.3.0-build138`.

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5584">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
